orchestrator/api/api_v1/endpoints/subscriptions.py

```diff
@@ -181,7 +185,14 @@ def subscription_workflows_by_id(subscription_id: UUID) -> dict[str, list[dict[s
     if not subscription:
         raise_status(HTTPStatus.NOT_FOUND)
 
-    return subscription_workflows(subscription)
+    subscription_workflows_dict = subscription_workflows(subscription)
+    for workflow_target in Target.values():
+        for workflow_dict in subscription_workflows_dict[workflow_target.lower()]:
+            workflow = get_workflow(workflow_dict["name"])
+            if workflow.authorize_callback and not workflow.authorize_callback(current_user):  # type: ignore
+                workflow_dict["reason"] = "subscription.insufficient_workflow_permissions"
```

Comment:
Would be nice to put this in a function `authorized_subscription_workflows()` or something (in that same services/subscriptions.py module) which takes the OIDCUserModel as parameter and returns this as a result. Keeps the endpoint code clean and makes it a little less work to reuse in the future (which I can imagine we will)

Also some comments on the code itself;
- If the inner for-loop is made to iterate over `subscription_workflows_dict.values()` then I think you can get rid of the outer for-loop
- `workflow_dict["reason"]` can already be set, for example if the subscription is out of sync. In this case we don't want to override this